File: app/src/main/java/com/elementary/tasks/notes/preview/PhotoPagerAdapter.java
package com.elementary.tasks.notes.preview;

import android.content.Context;

import androidx.annotation.NonNull;
import androidx.viewpager.widget.PagerAdapter;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;

import com.bumptech.glide.Glide;
import com.elementary.tasks.databinding.FragmentImageBinding;
import com.elementary.tasks.notes.create.NoteImage;

import java.util.List;

import uk.co.senab.photoview.PhotoView;
import uk.co.senab.photoview.PhotoViewAttacher;

class PhotoPagerAdapter extends PagerAdapter {

    private LayoutInflater mLayoutInflater;
    private List<NoteImage> mPhotosUrl;


    PhotoPagerAdapter(Context context, List<NoteImage> images) {
        mLayoutInflater = (LayoutInflater) context.getSystemService(Context.LAYOUT_INFLATER_SERVICE);
        mPhotosUrl = images;
    }

    @Override
    public int getCount() {
        return mPhotosUrl.size();
    }

    @Override
    public boolean isViewFromObject(@NonNull View view, @NonNull Object object) {
        return view.equals(object);
    }

    @NonNull
    @Override
    public Object instantiateItem(@NonNull ViewGroup container, int position) {
        FragmentImageBinding binding = FragmentImageBinding.inflate(mLayoutInflater, container, false);
        ImageView imageView = binding.ivPhoto;
        loadPhoto(imageView, position);
        PhotoViewAttacher mAttacher = new PhotoViewAttacher(imageView);
        mAttacher.setScaleType(ImageView.ScaleType.CENTER_INSIDE);
        container.addView(binding.getRoot());
        return binding.getRoot();
    }

    @Override
    public void destroyItem(@NonNull ViewGroup container, int position, @NonNull Object object) {
        container.removeView((PhotoView) object);
    }

    private void loadPhoto(ImageView imageView, int position) {
        NoteImage image = mPhotosUrl.get(position);
        Glide.with(imageView.getContext().getApplicationContext())
                .load(image.getImage())
                .into(imageView);
    }
}

File: app/src/main/java/com/elementary/tasks/notes/preview/PhotoPagerAdapter.kt
package com.elementary.tasks.notes.preview

import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ImageView
import androidx.viewpager.widget.PagerAdapter
import coil.load
import com.elementary.tasks.core.data.ui.note.UiNoteImage
import com.elementary.tasks.databinding.FragmentImageBinding
import com.github.chrisbanes.photoview.PhotoView

class PhotoPagerAdapter(private val images: List<UiNoteImage>) : PagerAdapter() {

  override fun getCount(): Int {
    return images.size
  }

  override fun isViewFromObject(view: View, `object`: Any): Boolean {
    return view == `object`
  }

  override fun instantiateItem(container: ViewGroup, position: Int): Any {
    val binding = FragmentImageBinding.inflate(
      LayoutInflater.from(container.context),
      container,
      false
    )
    loadPhoto(binding.ivPhoto, position)
    container.addView(binding.root)
    return binding.root
  }

  override fun destroyItem(container: ViewGroup, position: Int, `object`: Any) {
    container.removeView(`object` as PhotoView)
  }

  private fun loadPhoto(imageView: ImageView, position: Int) {
    imageView.load(images[position].filePath)
  }
}
